### ISSUE TITLE
Some shortcuts and Shortcut assignment not working properly

### ISSUE THREAD

**Issue by shaeriz** [zen-browser/desktop#1028]
### What happened?

I was trying to figure out why I couldn't unsplit tabs when I tried to assign a different keyboard shortcut to it, but when I use 3 key shortcuts it goes back to ''not set'' and becomes blank

![image](https://github.com/user-attachments/assets/70c1adcc-485c-45b7-affa-95517936e6b7)

and if I try 4+ keys shorcuts it'll get weird like this

![image](https://github.com/user-attachments/assets/9211ac7f-c60b-43d4-8588-747bf88abb3b)

I didn't find a 'restore to default' option either. Currently using latest Win 11 version (with a 60% keyboard)

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.0-a.32

### Severity impact

Medium

### What platform are you seeing the problem on?

Windows

**Comment by rve0** [comment]
Confirming that I'm experiencing the same behavior. Would like to add that some more details that I've found out:
1) Using Ctrl only instead of Ctrl+Alt will give you a shortcut with a lower case letter. Once you've assigned a lower case letter, the shortcut works.
2) Adding Shift to a Ctrl+Alt shortcut will product the 'Shift+¢' input. This is of course visually incorrect, but the shortcut will work as intended.

**Comment by rve0** [comment]
Dove a bit deeper and the way to change the upper case letters to lower case letters. I hoped that this would make Ctrl+Alt+C work without Shift. It unfortunately doesn't. The good news is that I've now got a solid grasp of what needs to happen and I've got some extra time on my hands so I'll see if I can fix this.

**Comment by rve0** [comment]
> stunning that the default keyboard shortcut doesn't work though. I guess it's still in alpha, but still. Thanks for looking into it @rve0 ! if it might help to find what causes the shortcuts to not work: ctrl+alt+U also doesnt work (adding the shift makes it work) for closing a plit view (unsplit). also notice that all the ctrl+alt+somekey shortcuts that do not work, are the same that become 'unset' when trying to rebind them. I guess this is most likely the same cause.

Yeah I bumped into this piece of code earlier. Likely plays a part ;) 

```js 
 _addDefaultShortcuts() {
    for (let action in kZenDefaultShortcuts) {
      if (!this._savedShortcuts[action]) {
        this._savedShortcuts[action] = this._parseDefaultShortcut(kZenDefaultShortcuts[action]);
      }
    }
  },
  ```
  
  But at the moment Im looking into the proper way of handling keyboard events. Thats where things get nasty because there's a whole lot different types of keyboards that I have to consider. Luckily there's specs out there to guide me, but they're quite lengthy haha.

**Comment by rve0** [comment]
> Adding onto this, trying to assign function-row shortcuts leaves the field blank entirely and does not work. (F5 for refresh, F12 for dev tools, etc)

Yeah, this does happen. The root cause of this is slightly different from text being changed in this ticket, but it fits in the bigger scope of what should happen when keys are being pressed.

The root cause of Ctrl+Shift+Alt+C being changed is the OS sending a different event.key value. with event.keyCode being deprecated, it may be better to use event.code. There's some exciting implications that this may have (Left shift and right shift could be two different keys, so itll greatly increase the amount of possible shortcuts). 

I'll follow up with an overview in a second. I haven't fully tested the function row keys, so I'll include that first.

**Comment by Gustavo-DMS** [comment]
If i had to guess the problem might be related to the Ctrl + Alt  combo producing other key-codes on some keyboard languages. For example Ctrl + Alt + U on windows with the United States-International keyboard layout resolves to "ú". This seem to be the issue with me so it might be what you are dealing with, try to check the behavior of Altgr on your layout.

**Comment by rve0** [comment]
Yeah, using US international breaks the functionality, but US is working fine.

Since I forgot to include the overview;
US international:
![image](https://github.com/user-attachments/assets/29e392a6-aceb-4d44-814e-cb498f4c091c)

UK Qwerty:
![image](https://github.com/user-attachments/assets/05146bbb-4b1e-45b2-8673-4a2bbabd1a6b)

**Comment by rve0** [comment]
@mauro-balades this ticket can be closed since #2198 is about the same issue.

**Comment by BruceBufferOverflow** [comment]
> @mauro-balades this ticket can be closed since #2198 is about the same issue.

Why should a duplicate of an issue be kept and not the original posting? Dupes should be the ones to be closed/removed.

**Comment by mr-cheffy** [comment]
True, but I'll do change it as the other one is more recent and more up to date
